plugins/lsp/default.nix

```diff
@@ -231,34 +260,37 @@ in {
     mkIf cfg.enable {
       extraPlugins = [pkgs.vimPlugins.nvim-lspconfig];
 
-      keymaps = let
-        mkMaps = prefix:
-          mapAttrsToList
-          (
-            key: action: let
-              actionStr =
-                if isString action
-                then action
-                else action.action;
-              actionProps =
-                if isString action
-                then {}
-                else filterAttrs (n: v: n != "action") action;
-            in {
-              mode = "n";
-              inherit key;
-              action.__raw = prefix + actionStr;
+      keymapsOnEvents = {
+        "LspAttach" = let
+          mkMaps = prefix:
+            mapAttrsToList
+            (
+              key: action: let
+                actionStr =
+                  if isString action
+                  then action
+                  else action.action;
+                actionProps =
+                  if isString action
+                  then {}
+                  else filterAttrs (n: v: n != "action") action;
+              in {
+                mode = "n";
+                inherit key;
+                action.__raw = prefix + actionStr;
```

Comment:
If this line is being indented anyway, is it worth changing it to use `helpers.mkRaw`?